index.go

```diff
@@ -203,10 +207,10 @@ func (w *indexWriter) write(bufs ...[]byte) error {
 	return nil
 }
 
-// addPadding adds zero byte padding until the file size is a multiple of n.
-func (w *indexWriter) addPadding(n int) error {
-	p := n - (int(w.pos) % n)
-	if p == 0 {
+// addPadding adds zero byte padding until the file size is a multiple size_unit.
+func (w *indexWriter) addPadding() error {
+	p := size_unit - (int(w.pos) & padding_and)
+	if p == size_unit {
```

Comment:
Would this be potentially simpler?

```go
p := w.pos % size_unit
if p == 0 {
    return nil
}
p = size_unit - p
```

That would skip the extra `padding_and` constant – `&` always need a few looks when reading it without context.

Reply:
Okay. Will do the change. % is sometimes expensive, but anyway 4 is a very small number.

Reply:
It will probably be optimized into `&` by the compiler – but we are not calling too often in the grand scheme of things anyway :)